### DISCUSSION TITLE
Does DevContainer support multiple repos?

### DISCUSSION THREAD

**Discussion by tigerinus** [devcontainers/spec#107] [Q&A]
Hello.

So far the DevContainer examples I have seen and attached to a Git repo each.

I'd like to create a DevContainer where people see multiple git repos when open in vscode, i.e. workspace.

Is this possible?

**Comment by bamurtaugh** [comment]
Thanks for opening @tigerinus. I think this feature request may address your scenario: https://github.com/microsoft/vscode-remote-release/issues/3665. Please let me know if it is different or you have other thoughts, thanks!

**Reply by Chuxel** [reply]
There's nothing specific about the dev container spec that prevents the use of multiple repositories. That said, there are some tips for VS Code and Codespaces since you mentioned a specific tool here:

* Once you are in a Dev Container in VS Code, the File > Open Workspace... menu item will let you select a .code-workspace file in the filesystem.
* If you don't already have one, you can create a "bootstrap" repository to create your initial environment, and then clone multiple repositories from there.
* There are also some specific properties for services like Codespaces to control access to other repositories you happen to be using that: https://docs.github.com/en/codespaces/managing-your-codespaces/managing-repository-access-for-your-codespaces

Is that what you were getting at or did you have something else in mind?

**Reply by tigerinus** [reply]
Yes, I am sure these steps would work after all.

The project I am working on is open source and involves multiple git repos when coding and debugging (search CasaOS in GitHub). Currently the bar to get someone from community to start contributing it somewhat high due to the setup. My goal is to have a working development environment within fewest steps, so anyone can write their first line of code to our project and see the change right away and gain confidence and trust, therefore continue to contribute.
